### ISSUE TITLE
Implement medals section in user profile overlay

### ISSUE THREAD

**Issue by frenzibyte** [ppy/osu#27487]
As medals are gradually becoming supported in osu!lazer, this is probably a good time to track adding the medals section in user profile overlay, for the user to be able to view all the medals they have unlocked, without having to go through the web.

Opening issue thread in case it's not been planned already. @bdach you might be interested in this one, but I can look into it myself to lower the effort if that's preferred.

**Comment by bdach** [comment]
Yes I am interested but it is also literally not possible to implement this right now because there is no sane way to receive this data from web. [See relevant discord conversation.](https://discord.com/channels/188630481301012481/188630616286167041/1209514189362692149).

@ppy/team-web please advise how this could be implemented.

**Comment by nanaya** [comment]
can add an endpoint returning available medals

**Comment by nanaya** [comment]
There's currently no interface to return all medals, including locked, of a user
